### PR TITLE
Andy Reid Homework 3

### PR DESCRIPTION
**Pull request by wildalmondz**
I tried to incorporate the default date to null and you'll see I ran into some challenges. I turned in my working code.

Thanks

--Andy

### CODE REVIEW

File: enhanced-date.js [hunk 1]
```diff
@@ -1 +1,113 @@
...
+}
+
+var checkDate = mm + dd + yyyy;
+var formatDate = mm + '/' + dd + '/' + yyyy;
+
+module.exports = {
+
+  setDate: function(newDate) {  // make a function for milli
+
+    if (privateDate == null) {
```
Comment (jamdon) [inline enhanced-date.js:25]: use `===` in favor of `==`

File: enhanced-date.js [hunk 2]
```diff
@@ -1 +1,113 @@
...
+    }
+
+    else {  // only using newMilli in use
+      privateDate = new Date();
+      console.log(privateDate);
+    }
+  },
+
+  getDate: function(dateAsk) {
+    if (dateAsk == undefined) {
```
Comment (jamdon) [inline enhanced-date.js:38]: use `===` in favor of `==`

File: enhanced-date.js [hunk 3]
```diff
@@ -1 +1,113 @@
...
+      console.log('No Args');
+      var n = privateDate.getTime();
+
+      return n;
+    }
+    else if (dateAsk == 'milliseconds') {
```
Comment (jamdon) [inline enhanced-date.js:44]: use `===` in favor of `==`

File: enhanced-date.js [hunk 4]
```diff
@@ -1 +1,113 @@
...
+      console.log('milliseconds');
+      var m = privateDate.getTime();
+      console.log(m);
+    }
+    else if (dateAsk == 'formatted') {
```
Comment (jamdon) [inline enhanced-date.js:49]: use `===` in favor of `==`

File: enhanced-date.js [hunk 5]
```diff
@@ -1 +1,113 @@
...
+      console.log('given date is not in the future');
+    }
+    else {
+      console.log('date is in the future');
+    }
+  },
+
+  isToday: function(dateAsk) {
+
+    if (dateAsk == checkDate) {
```
Comment (jamdon) [inline enhanced-date.js:105]: use `===` in favor of `==`

File: enhanced-date_2.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+  checkDate = mm + dd + yyyy;
+  formatDate = mm + '/' + dd + '/' + yyyy;
+}
+
+module.exports = {
+
+  setDate: function(newDate) {  // make a function for milli
+
+    if (privateDate == null) {
```
Comment (jamdon) [inline enhanced-date_2.js:37]: use `===` in favor of `==`

File: enhanced-date_2.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    else {  // only using newMilli in use
+      privateDate = new Date();
+      console.log(privateDate);
+    }
+  },
+
+  getDate: function(dateAsk) {
+    if (dateAsk == undefined) {
```
Comment (jamdon) [inline enhanced-date_2.js:50]: use `===` in favor of `==`

File: enhanced-date_2.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+      console.log('No Args');
+      var n = privateDate.getTime();
+
+      return n;
+    }
+    else if (dateAsk == 'milliseconds') {
```
Comment (jamdon) [inline enhanced-date_2.js:56]: use `===` in favor of `==`

File: enhanced-date_2.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      console.log('milliseconds');
+      var m = privateDate.getTime();
+      console.log(m);
+    }
+    else if (dateAsk == 'formatted') {
```
Comment (jamdon) [inline enhanced-date_2.js:61]: use `===` in favor of `==`

File: enhanced-date_2.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+      weekday[5] = 'Friday';
+      weekday[6] = 'Saturday';
+      console.log(weekday[formatDay]);
+
+      return (weekday[formatDay]);
+    }
+  },
+
+  getMonthName: function(dateAsk) {
+    if ((dateAsk) && (privateDate == null)) { // get by the "Can't read property of 'null' error
```
Comment (jamdon) [inline enhanced-date_2.js:91]: use `===` in favor of `==`

File: enhanced-date_2.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+      console.log('given date is not in the future');
+    }
+    else {
+      console.log('date is in the future');
+    }
+  },
+
+  isToday: function(dateAsk) {
+
+    if (dateAsk == checkDate) {
```
Comment (jamdon) [inline enhanced-date_2.js:128]: use `===` in favor of `==`

File: trucks.js [hunk 12]
```diff
@@ -1,256 +1,290 @@
...
+    function checkTruck(testArray) {
+      // element of Food Truck array of days vs. testArray
+      return testArray == day;
+    }
+
+    for (var i = 0; i < foodTrucks.length; i++) {
+      testArray = foodTrucks[i].schedule;
+      var truckStatus = testArray.filter(checkTruck);
+
+      if (truckStatus == day) {
```
Comment (jamdon) [inline trucks.js:280]: use `===` in favor of `==`

File: trucks.js [hunk 13]
```diff
@@ -1,256 +1,290 @@
...
+
+module.exports = {
+
+  filterByDay: function (day) {
+    var testArray;
+    var trucksToday = [];
+
+    function checkTruck(testArray) {
+      // element of Food Truck array of days vs. testArray
+      return testArray == day;
```
Comment (jamdon) [inline trucks.js:273]: use `===` in favor of `==`

File: trucks.js [hunk 14]
```diff
@@ -1,256 +1,290 @@
...
\ No newline at end of file
+ */
+
+module.exports = {
+
+  filterByDay: function (day) {
+    var testArray;
+    var trucksToday = [];
+
+    function checkTruck(testArray) {
```
Comment (jamdon) [inline trucks.js:271]: Avoid declaring functions inside other functions like this. Additionally, avoid the use of the same variable name used for different purposes within the same part of code - in this case `testArray` as it makes the code more difficult to follow.

**Comment by jamdon** [comment]
Works good. I've left some inline comments regarding best practices. 👍

File: enhanced-date_2.js (hunk 11 above)
Comment (wildalmondz) [inline enhanced-date_2.js:128]: Will do, my bad I missed these previous messages. I should have spent some time with gitHub and learned how to get these notes.  I'll adjust for Homework 4, thanks James.

--Andy

**Pull request closed without merging**